src/values/phi_value.rs

```diff
@@ -81,7 +81,7 @@ impl<'ctx> PhiValue<'ctx> {
         self.phi_value.as_instruction().expect("PhiValue should always be a Phi InstructionValue")
     }
 
-    pub fn replace_all_uses_with(self, other: &PhiValue<'ctx>) {
+    pub fn replace_all_uses_with(self, other: &dyn AnyValue<'ctx>) {
```

Comment:
Would you be able to write tests to verify that various AnyValue types are valid here?

Also, we should turn this into a generic rather than a dyn trait.

Reply:
Sorry for the late response. I added a test and turned it into a generic function.